cs/src/Management/TunnelManagementClient.cs

```diff
@@ -400,10 +400,12 @@ private static void ValidateHttpHandler(HttpMessageHandler httpHandler)
                     case HttpStatusCode.Forbidden:
                         var ex = new UnauthorizedAccessException(errorMessage, hrex);
 
-                        if (response.Headers.WwwAuthenticate?.Count > 0)
+                        // The HttpResponseHeaders.WwwAuthenticate property does not correctly
```

Comment:
I merged in main to my branch to avoid conflicts. If main is merged to your dev branch, these commits will go away. But I can retarget this to `main` if you're okay with that for a 'beta' client 🙂

Reply:
Yes I think it's fine for this to merge into `main` even though it is incomplete, since the part that's done is useful. You/we can iterate on it further there.

Please re-target to `main` and then I'll approve the PR.